app/environments/brassbirmingham/brassbirmingham/envs/classes/board.py

```diff
@@ -0,0 +1,538 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, List, Optional
+
+from consts import (
+    CANAL_PRICE,
+    ONE_RAILROAD_COAL_PRICE,
+    ONE_RAILROAD_PRICE,
+    ROAD_LOCATIONS,
+    STARTING_CARDS,
+    STARTING_WILD_BUILDING_CARDS,
+    STARTING_WILD_LOCATION_CARDS,
+    TOWNS,
+    TRADEPOSTS,
+    TWO_RAILROAD_COAL_PRICE,
+    TWO_RAILROAD_PRICE,
+    MAX_MARKET_COAL,
+    MAX_MARKET_IRON
+)
+from python.id import id
+from python.print_colors import *
+
+from .build_location import BuildLocation
+from .buildings.building import Building
+from .buildings.enums import BuildingName, BuildingType
+from .deck import Deck
+from .road_location import RoadLocation
+from .roads.canal import Canal
+from .roads.railroad import Railroad
+from .town import Town
+from .trade_post import TradePost
+
+if TYPE_CHECKING:
+    from .player import Player
+
+
+class Board:
+    def __init__(self, numPlayers: int):
+        self.id = id()
+        self.deck = Deck(STARTING_CARDS[str(numPlayers)])
+        self.wildBuildingDeck = Deck(STARTING_WILD_BUILDING_CARDS)
+        self.wildLocationDeck = Deck(STARTING_WILD_LOCATION_CARDS)
+        self.towns = TOWNS  # array of Town objects
+        self.townMap = {}
```

Comment:
nit: Normally I think of `map` as `dict`, but in this case because of the fact that `townMap` could mean something different b/c of the `town` prefix, maybe we should rename this to `townDict` or something similar?